### ISSUE TITLE
Consider option to open preview in a separate window

### ISSUE THREAD

**Issue by vixalien** [sonnyp/Workbench#270]
It would be nice to have an option to open the preview in a separate window.

This can be useful when testing a responsive layout and seeing how the UI reacts to different sizes.

**Comment by sonnyp** [comment]
:+1: 

hint: if you don't have JS code, you can do so by selecting Vala instead of JavaScript

**Comment by sonnyp** [comment]
Here are some pointers to get this working

There is a menu now in the preview panel so we can add the option "Preview in Window" there using `settings.create_action`.

As for actually opening the preview in a window,  it can be done here https://github.com/sonnyp/Workbench/blob/fa40deb025adb1a78a6b853ec00ae4c9ce1678f4/src/Previewer/Internal.js#L73

**Comment by vixalien** [comment]
thanks. how does vala do it though, I think `preview()` just sets the child.

oh, does that mean we would need to create a new `GtkWindow` if the Preview in Window option is enabled??

**Comment by sonnyp** [comment]
> oh, does that mean we would need to create a new GtkWindow if the "Preview in Window" option is enabled??

Right, we need to create a window and set its content with the object to preview. 
Note that the window needs to stay around so that it can be reused when there are updates.

**Comment by sonnyp** [comment]
An other option is to wrap your `UI` in a `GtkWindow` - Workbench will automagically offer to open the preview in a window. update works too.

But it would be nice if it worked out of the box.

**Comment by vixalien** [comment]
> But it would be nice if it worked without having to wrap.

for sure.

> Note that the window needs to stay around

I'm currently lost on how to implement this. is there some code that checks if the child is a `GtkWindow` or one of it's descendants, so that it offers to open the window in a separate window?

**Comment by sonnyp** [comment]
Search for `GtkRoot` in src/Previewer/utils.js

**Comment by vixalien** [comment]
hmm i can look at this at a later date. you can assign me if you like.
